### DISCUSSION TITLE
Duda tiempo de ejecucion

### DISCUSSION THREAD

**Discussion by waknaudt** [IIC2133-PUC/2024-2#34] [Tarea 0] [answered]
Hola, tengo 2 consultas sobre el tiempo de ejecucion:

- El tiempo de ejecucion se considera usando `time {programa}` o `time valgrind {programa}`? Porque el primero me demora a lo mas 1.5s en los test mas dificiles, mientras el segundo llega hasta los 15s en el mismo escenario
- Lo otro es que como el tiempo de ejecucion puede variar segun hardware, esto me tiene intranquilo. Como se si se demorara lo mismo aprox en el computado donde sera corregido?

**Comment by 4n1t4** [comment, marked as answer]
Respecto al comando, debes considerar lo siguiente:
* time: Mide el tiempo real de ejecución de un programa sin overhead significativo.

* Valgrind time: Mide el tiempo de ejecución mientras el programa está siendo analizado exhaustivamente por Valgrind, lo que introduce un overhead considerable.

Por ende el que realmente muestra cuanto se demora el programa es Time. Este debe cumplir el requisito de tiempo. 

Respecto a lo según generalmente en el servidor que corremos la tarea no existe demasiada variación de tiempo, sin embargo siempre podemos hacer correcciones en donde manda generalmente el tiempo que te sale en tu maquina local.

Espero quede más claro

**Reply by waknaudt** [reply]
Gracias!
